lib/money/money.rb

```diff
@@ -1,77 +1,107 @@
-require 'bigdecimal'
-require 'bigdecimal/util'
-
 class Money
   include Comparable
 
-  attr_reader :value, :cents
+  attr_reader :value, :cents, :currency
 
   class << self
-    def new(value = 0, _currency = nil)
+    attr_accessor :parser, :default_currency
+
+    def new(value = 0, currency = nil)
+      currency ||= default_currency
+
       if value == 0
-        @empty ||= super(0)
+        @@zero_money ||= {}
+        @@zero_money[currency] ||= super(0, currency)
       else
-        super(value, _currency)
+        super(value, currency)
       end
     end
+    alias_method :from_amount, :new
 
-    def empty
-      new(0)
+    def zero(currency = nil)
+      new(0, currency)
     end
+    alias_method :empty, :zero
 
-    alias_method :from_amount, :new
+    def parse(input, _currency = nil)
+      parser.parse(input)
+    end
+
+    def from_cents(cents, currency = nil)
+      new(cents.round.to_f / 100, currency)
+    end
+
+    def from_subunits(subunits, currency_iso)
+      currency = Currency.find!(currency_iso)
+      value = Helpers.value_to_decimal(subunits) / currency.subunit_to_unit
+      new(value, currency)
+    end
+
+    def default_settings
+      self.parser = MoneyParser
+      self.default_currency = 'xxx'
```

Comment:
`'xxx'` looks like a magic constant. Is this used to represent an unknown currency?